src/main/java/io/split/android/client/storage/mysegments/SqLitePersistentMySegmentsStorage.java

```diff
@@ -18,43 +16,39 @@
 
 public class SqLitePersistentMySegmentsStorage implements PersistentMySegmentsStorage {
 
-    private static final Type MY_SEGMENTS_LIST_TYPE = new TypeToken<List<String>>() {
-    }.getType();
     final SplitRoomDatabase mDatabase;
-    final String mUserKey;
     final StringHelper mStringHelper;
 
-    public SqLitePersistentMySegmentsStorage(@NonNull SplitRoomDatabase database, @NonNull String userKey) {
+    public SqLitePersistentMySegmentsStorage(@NonNull SplitRoomDatabase database) {
         mDatabase = checkNotNull(database);
-        mUserKey = checkNotNull(userKey);
         mStringHelper = new StringHelper();
     }
 
     @Override
-    public void set(List<String> mySegments) {
-        if(mySegments == null) {
+    public void set(String userKey, @NonNull List<String> mySegments) {
+        if (mySegments == null) {
             return;
         }
         MySegmentEntity entity = new MySegmentEntity();
-        entity.setUserKey(mUserKey);
+        entity.setUserKey(userKey);
         entity.setSegmentList(mStringHelper.join(",", mySegments));
         entity.setUpdatedAt(System.currentTimeMillis() / 1000);
         mDatabase.mySegmentDao().update(entity);
     }
 
     @Override
-    public List<String> getSnapshot() {
-        return getMySegmentsFromEntity(mDatabase.mySegmentDao().getByUserKeys(mUserKey));
+    public List<String> getSnapshot(String userKey) {
+        return getMySegmentsFromEntity(mDatabase.mySegmentDao().getByUserKeys(userKey));
```

Comment:
Assuming there's not a lot of risk, can we rename this method `getByUserKeys` from the DAO to be singular? Which is what the query does if I read it correctly. It can most definitely be a change at the polishing phase.